sky/packages/sky/lib/rendering/object.dart

```diff
@@ -522,9 +651,14 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
         context.canvas.save();
         context.canvas.clipRect(paintBounds.shift(offset));
       }
+      if (hasLayer)
+        assert(_layer != null);
```

Comment:
Done. In case it affects your opinion, note that this whole block is inside another assert(); my original thought was that using the if() here was slightly clearer and didn't matter since it was all in an assert. But I agree that this is more idiomatic for asserts in general.